swap_meet/vendor.py

```diff
@@ -1,2 +1,118 @@
+
+    def add(self, add_item):
+        '''
+        It adds item to the inventory.
+        Argument - add_item(string): representing an item
+        Return: the item that was added 
+        '''
+        self.inventory.append(add_item)
+        return add_item
+
+    def remove(self, remove_item):
+        '''
+        It removes the matching item from the inventory.
+        Argument - remove_item(string): representing an item
+        Return: the item that was removed, if there is no matching item it returns False
+        '''
+        if remove_item in self.inventory:
+            self.inventory.remove(remove_item)
+            return remove_item
+        else:
+            return False
+
+    def get_by_category(self, category):
+        '''
+        It gets a list of the inventory and its category
+        Argument - category(string): representing a category
+        Return: a list of Items in the inventory with that category
+        '''
+        results = []
+        for item in self.inventory:
+            if category == item.category:
+                results.append(item)
+        return results
+
+    def swap_items(self, another_vendor, my_item, their_item):
+        '''
+        It swaps my item and my friend's item
+        Argument - 
+            another_vendor: the friend whom I(vendor but I will say I) am swapping my item with
+            my_item: the item I give to the friend
+            their_item: the item my friend gives to me
+        Return: if my inventory doesn't contain my_item or the friend's inventory doesn't contain their_item, it returns False, otherwise returns True
+        '''
+        if my_item not in self.inventory or their_item not in another_vendor.inventory:
+            return False
+        '''
+        Hi Audrey, I have two options here. 1. line 58 ~ 65, 2. line 68 ~ 71 
+        Which way would be a better way? :) 
+
+        another way of writing line  
+        my_inventory = self.inventory
+        friend_inventory = another_vendor.inventory
+
+        my_inventory.remove(my_item)
+        friend_inventory.append(my_item)
+        friend_inventory.remove(their_item)
+        my_inventory.append(their_item)
+        '''
+
+        self.inventory.remove(my_item)
+        another_vendor.inventory.append(my_item)
+        another_vendor.inventory.remove(their_item)
+        self.inventory.append(their_item)
+        return True
+    
+    def swap_first_item(self, another_vendor):
+        '''
+        A method of the class Vendor. It swaps both my and my friend's first item
+        Argument - another_vendor: the friend whom I am swapping my item with
+        Return: if either the friend or I have an empty inventory, it returns False, otherwise returns True
+        '''
+        if not another_vendor.inventory or not self.inventory:
+            return False
+        
+        self.swap_items(another_vendor, self.inventory[0], another_vendor.inventory[0])
+        return True
```

Comment:
Nice! Great use of `swap_items` as a helper method!

Reply:
Currently, swapping any items has a time complexity of O(n) due to the `remove` method in `swap_items`. We can make a subtle improvement to O(1) by using multiple-assignment instead:

```
        self.inventory[0], vendor.inventory[0] = vendor.inventory[0], self.inventory[0]
```        

Here's more info on Python multiple-assignment: https://www.w3schools.com/python/gloss_python_assign_value_to_multiple_variables.asp